src/components/HistoryTable/HistoryTableContext.tsx

```diff
@@ -26,10 +29,20 @@ interface HistoryTableState {
   columnLimit: number;
   historyTableFilters: TestFilter[];
   setHistoryTableFilters: (filters: TestFilter[]) => void;
+  setSelectedCommit: (order: number) => void;
+  markSelectedVisited: () => void;
+  selectedCommit: {
+    order: number;
+    loaded: boolean;
+    visited: boolean;
+    rowIndex: number;
```

Comment:
rowIndex and loaded could could be combined into the same variable if we calculate the rowIndex based on existing commits and not just new ones.